modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/publisher/APIApplicationPublisherImpl.java

```diff
@@ -46,31 +52,39 @@ public void publish(APIApplication apiApplication) {
 				"APIApplicationPublisher not available");
 		}
 
-		String osgiJaxRsName = _getOSGiJaxRsName(apiApplication);
+		_endpointMatchersMap.put(
+			_getEndpointMatcherKey(
+				apiApplication.getBaseURL(), apiApplication.getCompanyId()),
+			new EndpointMatcher(apiApplication.getEndpoints()));
 
-		_serviceRegistrationsMap.computeIfAbsent(
-			osgiJaxRsName,
-			key -> new ArrayList<ServiceRegistration<?>>() {
-				{
-					add(_registerApplication(apiApplication, osgiJaxRsName));
+		ServiceRegistration<Application> applicationServiceRegistration =
+			_applicationServiceRegistrationsMap.get(
```

Comment:
With this, we are mantaining two different maps. Maybe it could be done just by using the _serviceRegistrationsMap 🤔

If the value exists, then we need to update the properties of the application service registration; otherwise, we will register the application and resources. What if we use the already existing map instead of using a new one?

Reply:
But then we have to loop through the values and find the application... in the serviceRegistrationsMap we have also the resources.